--- a/libs/filesystem/v2/src/v2_operations.cpp
+++ b/libs/filesystem/v2/src/v2_operations.cpp
@@ -59,7 +59,13 @@
 # else // BOOST_POSIX_API
 #   include <sys/types.h>
 #   if !defined(__APPLE__) && !defined(__OpenBSD__)
+#     ifndef __ANDROID__
-#     include <sys/statvfs.h>
+#       include <sys/statvfs.h>
+#     else
+#       include <sys/vfs.h>
+#       define statvfs statfs
+#       define fstatvfs fstatfs
+#     endif
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
 #   else
